### PR TITLE
Add mode to parser to support $$ in pg

### LINKED ISSUE #15835: SQL parse error when execute postgresql select statement
### Which version of ShardingSphere did you use?

[b3e90fc](https://github.com/apache/shardingsphere/commit/b3e90fc9f8df3057b75ae3c7630a6c02b930d0d7)

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

ShardingSphere-Proxy

### Expected behavior

Execute `SELECT * FROM cypher('sharding_test_1', $$ CREATE (n) $$) as (a agtype)` successfully.

### Actual behavior

```
test=> SELECT * FROM cypher('sharding_test_1', $$ CREATE (n) $$) as (a agtype);
ERROR:  You have an error in your SQL syntax
test=> SELECT * FROM cypher('sharding_test_2', $$ CREATE (n) $$) as (a agtype);
ERROR:  You have an error in your SQL syntax
```

### PR DESCRIPTION
**Pull request by tuichenchuxin**
Fixes #15835.

Changes proposed in this pull request:
- add Add mode to parser to support $$ in pg
- add test case

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/imports/postgresql/ModeLexer.g4 [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+lexer grammar ModeLexer;
+
+import Symbol, PostgreSQLKeyword, Keyword, Comments, Literals;
+
+BeginDollarStringConstant
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/imports/postgresql/ModeLexer.g4:22]: @tuichenchuxin Please rename it with BEGIN_DOLLAR_STRING_CONSTANT.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/parser/PostgreSQLLexerBase.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+package org.apache.shardingsphere.sql.parser.postgresql.parser;
+
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.Lexer;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+public abstract class PostgreSQLLexerBase extends Lexer {
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/parser/PostgreSQLLexerBase.java:26]: Add javadoc for PostgreSQLLexerBase.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/parser/PostgreSQLLexerBase.java (hunk 2 above)
Comment (tuichenchuxin) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/parser/PostgreSQLLexerBase.java:29]: Thanks. Done.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/imports/postgresql/ModeLexer.g4 [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+   : ~ '$'+
+   | '$' ~ '$'*
+   ;
+
+EndDollarStringConstant
+   : ('$' Tag? '$')
+   {isTag()}?
+   {popTag();} -> popMode
+   ;
+
```
[the remark this answers is not in the document]
Comment (tuichenchuxin) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/imports/postgresql/ModeLexer.g4]: Thanks. Done.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**